### ISSUE TITLE
Error Fetching Environment

### ISSUE THREAD

**Issue by jny986** [appcelerator/vscode-appcelerator-titanium#484]
### Describe the Bug
Unable to get the android or iOs simulation devices from plugin

### Steps to reproduce the Bug
Steps to reproduce the behavior, e.g.:
1. Go to Titanium Plugin Toolbar tab
2. Click on build > iOs > simulator & build > Android > Emulator
3. See error:

`Cannot read property 'ios' of undefined`
`Cannot read property 'android' of undefined`
`Error fetching Appcelerator environment`

### Expected Behavior
I was previously able to see the simulators/emulators that I have setup on my machine

### Your Environment
| Software | Version |
| ---------------- | ---------- |
| Titanium SDK | 9.2.2.GA |
| Appcelerator CLI | 8.1.1 |
| VS Code | 1.50.1 d2e414d9e4239a252d1ab117bd7067f125afd80a x64 |
| vscode-titanium | 0.9.1 |

running on macOs Catalina 10.15.7

### Logs
I have not been able to get any logs

**Comment by ewanharris** [comment]
Hey @jny986. Our error logging for that failure is pretty awful, if you run `appc info` does it report any errors?

**Comment by jny986** [comment]
Hows it going @ewanharris.

Yeah I am using the latest Nodejs LTS 14.14.0

`appc info` does not show any errors

`appc ti info` shows the following error:
```
Appcelerator Command-Line Interface, version 8.1.1
Copyright (c) 2014-2020, Appcelerator, Inc.  All Rights Reserved.

 |(node:33100) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
 |An uncaught exception was thrown!
Rebuild failed:
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.10.3
node-pre-gyp info using node@14.14.0 | darwin | x64
node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp info check checked for "/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64/node_ios_device.node" (not found)
node-pre-gyp http GET https://appc-node-binaries.s3.amazonaws.com/node-ios-device/v1.7.1/node_ios_device-v1.7.1-node-v83-darwin-x64.tar.gz
node-pre-gyp http 403 https://appc-node-binaries.s3.amazonaws.com/node-ios-device/v1.7.1/node_ios_device-v1.7.1-node-v83-darwin-x64.tar.gz
node-pre-gyp WARN Tried to download(403): https://appc-node-binaries.s3.amazonaws.com/node-ios-device/v1.7.1/node_ios_device-v1.7.1-node-v83-darwin-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for node-ios-device@1.7.1 and node@14.14.0 (node-v83 ABI, unknown) (falling back to source compile with node-gyp) 
node-pre-gyp http 403 status code downloading tarball https://appc-node-binaries.s3.amazonaws.com/node-ios-device/v1.7.1/node_ios_device-v1.7.1-node-v83-darwin-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not installable for node-ios-device@1.7.1 and node@14.14.0 (node-v83 ABI, unknown) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Hit error Connection closed while downloading tarball file 
gyp info it worked if it ends with ok
gypgyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
 info it worked if it ends with ok
gyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
gypgyp  info infook 
 ok 
gyp info it worked if it ends with ok
gypgyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
 info it worked if it ends with ok
gyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
gyp info find Python using Python version 3.7.6 found at "/opt/miniconda3/bin/python"
gyp info find Python using Python version 3.7.6 found at "/opt/miniconda3/bin/python"
gyp info spawn /opt/miniconda3/bin/python
gyp info spawn args [
gyp info spawn args   '/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/Library/Caches/node-gyp/14.14.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/josh/Library/Caches/node-gyp/14.14.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/josh/Library/Caches/node-gyp/14.14.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn /opt/miniconda3/bin/python
gyp info spawn args [
gyp info spawn args   '/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/Library/Caches/node-gyp/14.14.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/josh/Library/Caches/node-gyp/14.14.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/josh/Library/Caches/node-gyp/14.14.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info ok 
gyp info ok 
gyp info it worked if it ends with ok
gyp info it worked if it ends with ok
gyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
gyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
clang: error: no such file or directory: 'Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/build'
clang: error: no such file or directory: 'Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/build'
make: *** [Release/obj.target/node_ios_device/src/device.o] Error 1make: 
*** [Release/obj.target/node_ios_device/src/device.o] Error 1
gypgyp ERR! build error  
ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/lib/build.js:194:23)
gyp gypERR!  stackERR!      at ChildProcess.emit (events.js:315:20)
stackgyp Error: `make` failed with exit code: 2
 ERR!gyp  stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
ERR! stack     at ChildProcess.onExit (/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Darwin 19.6.0
gypgyp  ERR!ERR!  commandSystem "/usr/local/Cellar/node/14.14.0/bin/node" "/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64/node_ios_device.node" "--module_name=node_ios_device" "--module_path=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64" "--napi_version=7" "--node_abi_napi=napi"
 Darwin 19.6.0
gyp ERR! cwdgyp /Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device
 ERR!gyp  commandERR! "/usr/local/Cellar/node/14.14.0/bin/node" "/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64/node_ios_device.node" "--module_name=node_ios_device" "--module_path=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64" "--napi_version=7" "--node_abi_napi=napi"
 node -v v14.14.0
gyp gypERR!  ERR!cwd  /Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device
node-gyp -v v5.1.0
gyp gypERR!  ERR!node -v  v14.14.0
not ok 
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/Cellar/node/14.14.0/bin/node /Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64/node_ios_device.node --module_name=node_ios_device --module_path=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64 --napi_version=7 --node_abi_napi=napi' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/node_modules/node-pre-gyp-init/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1048:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
node-pre-gyp ERR! System Darwin 19.6.0
node-pre-gyp ERR! command "/usr/local/Cellar/node/14.14.0/bin/node" "/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/node_modules/node-pre-gyp-init/node_modules/node-pre-gyp/bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device
node-pre-gyp ERR! node -v v14.14.0
node-pre-gyp ERR! node-pre-gyp -v v0.10.3
node-pre-gyp ERR! not ok
Rebuild failed:
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.10.3
node-pre-gyp info using node@14.14.0 | darwin | x64
node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp info check checked for "/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64/node_ios_device.node" (not found)
node-pre-gyp http GET https://appc-node-binaries.s3.amazonaws.com/node-ios-device/v1.7.1/node_ios_device-v1.7.1-node-v83-darwin-x64.tar.gz
node-pre-gyp http 403 https://appc-node-binaries.s3.amazonaws.com/node-ios-device/v1.7.1/node_ios_device-v1.7.1-node-v83-darwin-x64.tar.gz
node-pre-gyp WARN Tried to download(403): https://appc-node-binaries.s3.amazonaws.com/node-ios-device/v1.7.1/node_ios_device-v1.7.1-node-v83-darwin-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for node-ios-device@1.7.1 and node@14.14.0 (node-v83 ABI, unknown) (falling back to source compile with node-gyp) 
node-pre-gyp http 403 status code downloading tarball https://appc-node-binaries.s3.amazonaws.com/node-ios-device/v1.7.1/node_ios_device-v1.7.1-node-v83-darwin-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not installable for node-ios-device@1.7.1 and node@14.14.0 (node-v83 ABI, unknown) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Hit error Connection closed while downloading tarball file 
gyp info it worked if it ends with ok
gypgyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
 info it worked if it ends with ok
gyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
gypgyp  info infook 
 ok 
gyp info it worked if it ends with ok
gypgyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
 info it worked if it ends with ok
gyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
gyp info find Python using Python version 3.7.6 found at "/opt/miniconda3/bin/python"
gyp info find Python using Python version 3.7.6 found at "/opt/miniconda3/bin/python"
gyp info spawn /opt/miniconda3/bin/python
gyp info spawn args [
gyp info spawn args   '/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/Library/Caches/node-gyp/14.14.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/josh/Library/Caches/node-gyp/14.14.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/josh/Library/Caches/node-gyp/14.14.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn /opt/miniconda3/bin/python
gyp info spawn args [
gyp info spawn args   '/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/josh/Library/Caches/node-gyp/14.14.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/josh/Library/Caches/node-gyp/14.14.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/josh/Library/Caches/node-gyp/14.14.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info ok 
gyp info ok 
gyp info it worked if it ends with ok
gyp info it worked if it ends with ok
gyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
gyp info using node-gyp@5.1.0
gyp info using node@14.14.0 | darwin | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
clang: error: no such file or directory: 'Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/build'
clang: error: no such file or directory: 'Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/build'
make: *** [Release/obj.target/node_ios_device/src/device.o] Error 1make: 
*** [Release/obj.target/node_ios_device/src/device.o] Error 1
gypgyp ERR! build error  
ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/lib/build.js:194:23)
gyp gypERR!  stackERR!      at ChildProcess.emit (events.js:315:20)
stackgyp Error: `make` failed with exit code: 2
 ERR!gyp  stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
ERR! stack     at ChildProcess.onExit (/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Darwin 19.6.0
gypgyp  ERR!ERR!  commandSystem "/usr/local/Cellar/node/14.14.0/bin/node" "/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64/node_ios_device.node" "--module_name=node_ios_device" "--module_path=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64" "--napi_version=7" "--node_abi_napi=napi"
 Darwin 19.6.0
gyp ERR! cwdgyp /Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device
 ERR!gyp  commandERR! "/usr/local/Cellar/node/14.14.0/bin/node" "/Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64/node_ios_device.node" "--module_name=node_ios_device" "--module_path=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64" "--napi_version=7" "--node_abi_napi=napi"
 node -v v14.14.0
gyp gypERR!  ERR!cwd  /Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device
node-gyp -v v5.1.0
gyp gypERR!  ERR!node -v  v14.14.0
not ok 
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/Cellar/node/14.14.0/bin/node /Users/josh/.appcelerator/install/8.1.1/package/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64/node_ios_device.node --module_name=node_ios_device --module_path=/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/binding/node-v83-darwin-x64 --napi_version=7 --node_abi_napi=napi' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/node_modules/node-pre-gyp-init/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1048:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
node-pre-gyp ERR! System Darwin 19.6.0
node-pre-gyp ERR! command "/usr/local/Cellar/node/14.14.0/bin/node" "/Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device/node_modules/node-pre-gyp-init/node_modules/node-pre-gyp/bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/josh/Library/Application Support/Titanium/mobilesdk/osx/9.2.2.GA/node_modules/node-ios-device
node-pre-gyp ERR! node -v v14.14.0
node-pre-gyp ERR! node-pre-gyp -v v0.10.3
node-pre-gyp ERR! not ok
```
Any idea when you will be supporting the current Nodejs LTS?

**Comment by vlamoly** [comment]
Hi there ! 
I have exactly the same problem. 
I have almost the same configuration i.e

Software | Version
-- | --
Titanium SDK | 9.2.2.GA
Appcelerator CLI | 8.1.1
VS Code | 1.50.1 d2e414d9e4239a252d1ab117bd7067f125afd80a x64
vscode-titanium | 0.9.1

running on macOs Catalina 10.15.7

However I have installed **node v10.22.1** through nvm 

`appc info` and `appc ti info` do not show particular errors 
Except those issues below (but should not stop the environment fetching right ?). **Could it be a configured timeout because of the command taking too much time ?**
At first launch, in a new terminal, but once it is done, i don't have this anymore... except if i open a new terminal
```
You are attempting to run appc 8.1.1 which was compiled for node 8.9.1 but you are now running node v10.22.1
Rebuilding package modules ...
Package modules rebuilt!
```

And I also having those warning at the end 
```
Android Issues
  !  Unable to locate an Android NDK.
     Without the NDK, you will not be able to build native Android Titanium modules.
     If you have already downloaded and installed the Android NDK, you can tell Titanium where the
     Android NDK is located by running 'appc titanium config android.ndkPath /path/to/android-ndk',
     otherwise you can install it by running 'appc titanium setup android' or manually downloading
     from http://appcelerator.com/android-ndk.

  ✕  Unable to locate an Android SDK.
     If you have already downloaded and installed the Android SDK, you can tell Titanium where the
     Android SDK is located by running 'appc titanium config android.sdkPath /path/to/android-sdk',
     otherwise you can install it by running 'appc titanium setup android' or manually downloading
     from http://appcelerator.com/android-sdk.

iOS Issues
  !  Unable to find any valid iOS adhoc provisioning profiles.
     This will prevent you from packaging apps for adhoc distribution.
     You will need to log in to http://appcelerator.com/ios-dist-certs with your Apple Developer
     account, then create, download, and install a profile.
```

Thank you for your help
Vincent

**Comment by ewanharris** [comment]
@jny986, we'll be supporting Node 14 in the upcoming 9.3.0 release, it's available now by running `appc ti sdk install -b master -d`, that _should_ allow the info detection to work while your app still targets whatever the `sdk-version` in your tiapp is set to.

 
@vlamoly, do you receive that `Rebuilding package modules` frequently? If so it would indicate that something is running `appc` with Node 8.9.1 and therefore recompiling the modules. If that is VS Code then it might be the cause of the error 8.9.1 is unsupported and will cause the CLI to error. What does `nvm ls default` show is set as your default?

**Comment by ewanharris** [comment]
Closing out, please reopen a new issue if this persisted
